File: QTS/3° Bim/app-para-teste/desktop/src/main/java/br/edu/etec/biblioteca/models/Aluno.java
package br.edu.etec.biblioteca.models;

import br.edu.etec.biblioteca.core.Database;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.List;

/**
 * Model Aluno.
 * ATENÇÃO: código intencionalmente com FALHAS (SQL injection, validação frágil).
 */
public class Aluno {

    private final Connection conn;

    public Aluno() {
        this.conn = Database.getConnection();
    }

    /** Lista todos os alunos. */
    public List<String[]> todos() {
        List<String[]> lista = new ArrayList<>();
        try (Statement st = conn.createStatement();
             ResultSet rs = st.executeQuery("SELECT id, nome, email, turma, telefone FROM alunos")) {
            while (rs.next()) {
                lista.add(new String[]{
                    String.valueOf(rs.getInt("id")),
                    rs.getString("nome"),
                    rs.getString("email"),
                    rs.getString("turma"),
                    rs.getString("telefone")
                });
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
        return lista;
    }

    /**
     * Busca aluno por id.
     * BUG DE SEGURANÇA: SQL injection (id concatenado, não preparado).
     */
    public String[] buscar(int id) {
        String sql = "SELECT id, nome, email, turma, telefone FROM alunos WHERE id = " + id;
        try (Statement st = conn.createStatement();
             ResultSet rs = st.executeQuery(sql)) {
            if (rs.next()) {
                return new String[]{
                    String.valueOf(rs.getInt("id")),
                    rs.getString("nome"),
                    rs.getString("email"),
                    rs.getString("turma"),
                    rs.getString("telefone")
                };
            }
        } catch (SQLException e) {
            e.printStackTrace();
        }
        return null;
    }

    /**
     * Cria um aluno.
     * BUG DE SEGURANÇA: SQL injection (campos concatenados).
     * BUG DE VALIDAÇÃO: apenas verifica nome/email vazios; telefone/turma sem validar.
     */
    public boolean criar(String nome, String email, String turma, String telefone) {
        if (nome == null || nome.trim().isEmpty() || email == null || email.trim().isEmpty()) {
            // BUG: retorna false sem informar o usuário qual campo está errado.
            return false;
        }
        String sql = "INSERT INTO alunos (nome, email, turma, telefone) VALUES ("
            + "'" + nome + "', '" + email + "', '" + turma + "', '" + telefone + "')";
        try (Statement st = conn.createStatement()) {
            int linhas = st.executeUpdate(sql);
            // BUG LÓGICO: o resultado real (linhas) é ignorado; retorna true sempre
            // se não houve exceção, mesmo que nada tenha sido inserido.
            return true;
        } catch (SQLException e) {
            // BUG: exceção (ex.: email duplicado) vira apenas "false" sem mensagem.
            return false;
        }
    }

    /**
     * Remove um aluno pelo id.
     * BUG DE INTEGRIDADE: não verifica empréstimos pendentes antes de excluir.
     */
    public boolean remover(int id) {
        String sql = "DELETE FROM alunos WHERE id = " + id;
        try (Statement st = conn.createStatement()) {
            st.executeUpdate(sql);
            return true;
        } catch (SQLException e) {
            return false;
        }
    }

    /**
     * Valida e-mail.
     * BUG: retorna true mesmo para e-mails obviamente inválidos (validação fraca),
     * e não trata nulo (NullPointerException em chamadas com null).
     */
    public boolean emailValido(String email) {
        // Regex incompleta: "a@b" passaria sem ponto? Não — mas "a@@b.com.br" passa.
        if (email == null) {
            return false;
        }
        return email.contains("@") && email.contains(".");
    }
}

File: QTS/3° Bim/app-para-teste/web/app/models/Aluno.php
<?php
// Model Aluno
// ATENÇÃO: código intencionalmente com falhas para fins de estudo.

require_once __DIR__ . '/../core/Database.php';

class Aluno
{
    private $db;

    public function __construct()
    {
        $this->db = Database::getInstance();
    }

    // Obtém todos os alunos
    public function todos()
    {
        $resultado = $this->db->query("SELECT * FROM alunos");
        return $resultado->fetchAll(PDO::FETCH_ASSOC);
    }

    // Obtém um aluno pelo id
    public function buscar($id)
    {
        // BUG DE SEGURANÇA: SQL Injection - id não é parametrizado
        $sql = "SELECT * FROM alunos WHERE id = " . $id;
        $resultado = $this->db->query($sql);
        if ($resultado === false) {
            return null;
        }
        return $resultado->fetch(PDO::FETCH_ASSOC);
    }

    // Cria um novo aluno
    public function criar($nome, $email, $turma, $telefone)
    {
        // BUG: SQL Injection (email não sanitizado)
        // BUG LÓGICO: validar se os campos não são vazios está aqui, fora da camada de validação
        if ($nome == '' || $email == '') {
            return ['erro' => 'Nome e email são obrigatórios.'];
        }

        $sql = "INSERT INTO alunos (nome, email, turma, telefone)
                VALUES ('$nome', '$email', '$turma', '$telefone')";
        $this->db->exec($sql);

        // BUG: verifica se houve erro comparando com true (deveria ser !== false)
        // como exec retorna number of rows ou false, se inserir 1 linha, 1 == true é true
        $id = $this->db->getPdo()->lastInsertId();
        return ['id' => $id];
    }

    // Atualiza um aluno
    public function atualizar($id, $nome, $email, $turma, $telefone)
    {
        $sql = "UPDATE alunos SET
                    nome = '$nome',
                    email = '$email',
                    turma = '$turma',
                    telefone = '$telefone'
                WHERE id = $id";
        return $this->db->exec($sql);
    }

    // Remove um aluno
    public function remover($id)
    {
        $sql = "DELETE FROM alunos WHERE id = " . $id;
        // NÃO verifica se o aluno tem empréstimos pendentes -> perde integridade
        return $this->db->exec($sql);
    }

    // Validação de e-mail - reimplementação desnecessária e incompleta
    public function emailValido($email)
    {
        // BUG: nunca retorna true para emails comuns pois não considera domínio completo.
        // Ex.: ana@escola.edu não tem ponto no domínio -> filter_var retorna true, ok.
        if (!preg_match('/^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$/', $email)) {
            // BUG: mensagem de erro não informa qual campo falhou
            return 'E-mail inválido.';
        }
        return true;
    }
}
